code/modules/mob/living/carbon/xenomorph/life.dm

```diff
@@ -216,100 +218,25 @@
 		return FALSE
 	if(hud_used && hud_used.healths)
 		if(stat != DEAD)
-			switch(round(health * 100 / maxHealth)) //Maxhealth should never be zero or this will generate runtimes.
-				if(100 to INFINITY)
-					hud_used.healths.icon_state = "health_full"
-				if(94 to 99)
-					hud_used.healths.icon_state = "health_16"
-				if(88 to 93)
-					hud_used.healths.icon_state = "health_15"
-				if(82 to 87)
-					hud_used.healths.icon_state = "health_14"
-				if(76 to 81)
-					hud_used.healths.icon_state = "health_13"
-				if(70 to 75)
-					hud_used.healths.icon_state = "health_12"
-				if(64 to 69)
-					hud_used.healths.icon_state = "health_11"
-				if(58 to 63)
-					hud_used.healths.icon_state = "health_10"
-				if(52 to 57)
-					hud_used.healths.icon_state = "health_9"
-				if(46 to 51)
-					hud_used.healths.icon_state = "health_8"
-				if(40 to 45)
-					hud_used.healths.icon_state = "health_7"
-				if(34 to 39)
-					hud_used.healths.icon_state = "health_6"
-				if(28 to 33)
-					hud_used.healths.icon_state = "health_5"
-				if(22 to 27)
-					hud_used.healths.icon_state = "health_4"
-				if(16 to 21)
-					hud_used.healths.icon_state = "health_3"
-				if(10 to 15)
-					hud_used.healths.icon_state = "health_2"
-				if(4 to 9)
-					hud_used.healths.icon_state = "health_1"
-				if(0 to 3)
-					hud_used.healths.icon_state = "health_0"
-				else
-					hud_used.healths.icon_state = "health_critical"
+			if(maxHealth)
```

Comment:
If you want to do a check to avoid a division by zero, do it so it doesn't silently errors, as no mob with health should have `maxHealth`
Something like this would work:
```
if(!maxHealth)
	stack_trace("[src] called handle_regular_hud_updates() while having [maxHealth] maxHealth.")
	return
```
Will require to fix indentation.